gui/include/scanHelper.hh

```diff
@@ -0,0 +1,143 @@
+ * @abstract Store all parameters requried to perform a scan
+ * Additionally, the configs to be used are created and stored
+ *
+ */
+class Scan{
+public:
+    /**
+     * @brief The ScanSection struct, which stores the information from one user section in a Scan Configuration
+     */
+    struct ScanSection{
+        double start, step, stop, defaultV;
+        int secIndex;
+        bool nested;
+        string name, parameter, eventCounter;
+
+        ScanSection(double start, double stop, double step, string name,string parameter,string eventCounter, double defaultV, int idx, bool nested = false)
+            : start(start), stop(stop), step(step), name(name), nested(nested), parameter(parameter), eventCounter(eventCounter), defaultV(defaultV), secIndex(idx)
+        {}
+    };
+    Scan():m_allow_nested_scan(false), m_scan_is_time_based(true), m_time_per_step(0), m_repeatScans(false),
+        m_events_per_step(-10), m_current_step(0), m_config_files(), m_steps_per_scan(), m_n_steps(0),
+        m_scanned_parameter(), m_scanned_producer(), m_events_counting_component(), m_config_file_prefix("defaultScan")
+    {
+        reset();
+    }
+
+    /**
+     * @brief setupScan
+     * @param globalConfFile = default configuration file
+     * @param scanConfFile   = scan config file
+     * @return true if the scan could be configured correctly, false else
+     */
+    bool setupScan(string globalConfFile, string scanConfFile);
+    /**
+     * @brief Reset all local parameters, created files are not deleted
+     */
+    void reset();
+    /**
+     * @brief currentStep
+     * @return Current step in scan. Can be larger than the number of steps if a scan is started in an lopp
+     */
+    int currentStep()       const {return m_current_step;}
+    /**
+     * @brief nSteps
+     * @return number of steps in scan
+     */
+    int nSteps()            const {return m_n_steps;}
+
+    /**
+     * @brief timePerStep
+     * @return time per Step
+     */
+    int timePerStep()       const {return m_time_per_step;}
+    /**
+     * @brief eventsPerStep
+     * @return events per Step
+     */
+    int eventsPerStep()     const {return m_events_per_step;}
+
+    /**
+     * @brief nextConfig
+     * @return the configuration filename for the next step in the scan or 'finished', if the scan is completed
+     */
+    string nextConfig();
+    /**
+     * @brief scanIsTimeBased
+     * @return true if the steps are selected based on runtime
+     */
+    bool scanIsTimeBased()  const {return m_scan_is_time_based;}
+    /**
+     * @brief currentCountingComponent
+     * @return the connection which provides the number of events required in a scan step
+     */
+    string currentCountingComponent() const;
+private:
+
+    /**
+     * @brief ERROR message from EUDAQ_ERROR with return statement
+     * @param msg
+     * @return false
+     */
+    bool ERROR(string msg) {    std::cout << msg <<std::endl; EUDAQ_ERROR(msg); return false;}
```

Comment:
all up-case  ERROR is more like a pre-processing macro than function. Suggest to have some name else.

Reply:
done